modules/seqexec/server/src/main/scala/seqexec/server/CleanConfig.scala

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2016-2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.server
+
+import cats.implicits._
+import edu.gemini.spModel.config2.{Config, ItemEntry, ItemKey}
+import seqexec.server.ConfigUtilOps._
+import edu.gemini.spModel.gemini.gmos.InstGmosCommon.USE_NS_PROP
+import edu.gemini.spModel.obscomp.InstConstants.{ARC_OBSERVE_TYPE, FLAT_OBSERVE_TYPE, OBSERVE_TYPE_PROP, BIAS_OBSERVE_TYPE}
+import edu.gemini.spModel.seqcomp.SeqConfigNames.INSTRUMENT_KEY
+import seqexec.model.StepConfig
+import seqexec.model.enum.SystemName
+
+import scala.collection.breakOut
+
+/*
+ * CleanConfig is a wrapper over Config that allows to override some of the Config parameters. It allows to change some
+ * of the Config parameters without mutation.
+ * It is used to correct some inconsistencies in the sequences coming from the ODB. It has only one of those corrections
+ * now, but it allows for more to be added in the future.
+ */
+class CleanConfig(config: Config, overrides: Map[ItemKey, AnyRef]) {
```

Comment:
I wonder if this could be somehow `Map[ItemKey, _]`. I understand why it is `AnyRef` but now that we are going to the trouble of having `CleanConfig` we may try to tighten the screws

Perhaps it could be via a
`trait ItemValue[T]` and the map is `Map[ItemKey, ItemValue[_]]`?

I don't know. existentials give me the creeps. @tpolecat what do you think?

Reply:
I think we could `CleanConfig` client side but it means getting rid of all old ocs2 references. Would that be hard?

Reply:
My suggestion here is to add a check (it could just be a runtime assertion, we need to add a test to be sure it works) to ensure that the runtime type of overridden values are assignable to the same runtime types of the Config's values. So at least we can be confident that we're not turning a filter into an angle or something. I don't think types can really help us here because the underlying config is untyped.

Reply:
But all this will be solved when we move the Seqexec to the new OCS, right?

Right?